sections/docker/docker-ignore.md

```diff
@@ -0,0 +1,48 @@
+# Use .dockerignore to prevent leaking secrets
+
+<br/><br/>
+
+### One Paragraph Explainer
+
+The Docker build command copies the local files into the build context environment over a virtual network. Cautious, development and CI folders contain secrets like .npmrc, .aws, .env files, and other sensitive files. Consequently, Docker images might hold secrets and expose them in unsafe territories (e.g. Docker repository, partners servers). In a better world, the Dockerfile should be explicit about what is being copied. On top of this, include a .dockerignore file that acts as the last safety net that filters out unnecessary folders and potential secrets. Doing so also boosts the build speed - By leaving out common development folders that have no use in production (e.g. .git, test results, IDE configuration), the build can better utilize the cache and achieve better performance
+
+<br/><br/>
+
+### Code Example – A good default .dockerignore for Node.js apps
+
+<details>
+<summary><strong>.dockerignore</strong></summary>
+
+```
+**/node_modules/
+**/.git
+**/README.md
+**/LICENSE
+**/.vscode
+**/npm-debug.log
+**/coverage
+**/.env
+**/.editorconfig
+**/.aws
+```
+
+</details>
+
+<br/><br/>
+
+### Code Example Anti-Pattern – Recursive copy of all files
```

Comment:
We could add a Code example where we explicitly copy a `src` folder and an `assets` folder to better illustrate the `Dockerfile should be explicit about what is being copied` point

Reply:
@kevynb Wouldn't copying non-production assets confuse people?

Reply:
Well it can be a production asset if for example you need some img ressources or some csv files or whatever to load a list of supported products

Reply:
@kevynb Yes, good idea. I'm noting this down to the last mile enhancements. I prefer now to have a walking skeleton and then another iteration